lib/server.js

```diff
@@ -20,21 +20,20 @@
 var _ = require('lodash');
 var except = require('amoeba').except;
 var restify = require('restify');
-
+var errors = require('restify-errors');
 var log = require('./log.js')('lib/server.js');
 
 function resultsCb(request, response, next) {
   return function (err, results) {
     if (err != null) {
       log.info(err, 'Error on url[%s]', request.url);
       response.send(500);
+      return next(false);
     } else if (results == null) {
-      response.send(404);
-    } else {
-      response.send(200, results);
-    }
-
-    next();
+      return next(new errors.NotFoundError());
```

Comment:
Should this be sending the `errors.NotFoundError()` and returning `next(false)`?